crates/re_log_types/src/path/parse_path.rs

```diff
@@ -220,30 +225,26 @@ impl FromStr for ComponentPath {
 ///
 /// Things like `foo/Hallå Där!` will be accepted, and transformed into
 /// the path `foo/"Hallå Där!"`.
-fn parse_entity_path_forgiving(path: &str) -> Result<Vec<EntityPathPart>, PathParseError> {
-    if path.is_empty() {
-        return Err(PathParseError::EmptyString);
-    }
-
-    if path == "/" {
-        return Ok(vec![]); // special-case root entity
-    }
-
-    if path.starts_with('/') {
-        return Err(PathParseError::LeadingSlash);
-    }
-
+fn parse_entity_path_forgiving(path: &str) -> Vec<EntityPathPart> {
     let mut bytes = path.as_bytes();
```

Comment:
I didn't think of this in earlier reviews, but going by bytes here is quite dangerous. We should not go over bytes here and then do character comparisons - this may cause problems in the wild that are hard to debug. Instead, this should use `.chars()` everywhere here.

Reply:
oh, apologies I misremembered a thing. Here everything should be fine since all compared codepoints are uniquely identifiable from single bytes and there's thing that runs the risk of splitting characters.

Reply:
I added clarifying comment explaining why it is safe